packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx

```diff
@@ -34,6 +34,9 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
             ColumnMenuIcon: (props: SvgIconProps) => <MoreVertical {...props} fontSize="medium" />,
             ...component?.defaultProps?.components,
         },
+        localeText: {
+            noRowsLabel: "No results found",
```

Comment:
This string needs to be masked for i18n

Reply:
I doubt this is possible here. For i18n to work correctly, we probably need to override the no rows slot and use react-intl there.

An alternative we could evaluate would be reusing the DataGrid's label for no results, which already is "No results found." (note the punctuation): https://github.com/mui/mui-x/blob/HEAD/packages/grid/x-data-grid/src/constants/localeTextConstants.ts#L5-L6

Reply:
I left it like this because the DataGrid has it's own way of doing i18n (https://mui.com/x/react-data-grid/localization/). As far as I can see, we don't set the locale in the theme. So I guess the default DataGrid strings are never translated (unless I'm missing something).

We could override the slot to use react-intl. But then I also need to restyle the component. So I first opted for this simpler option to check what you think of it.

When I set 
```ts
localeText: {
     noRowsLabel: GRID_DEFAULT_LOCALE_TEXT.noResultsOverlayLabel,
},
```
the text is still overridden with the translation of `No rows` when the locale is set (e.g. `Keine Enträge` instead of `Keine Ergebnisse gefunden` in German).

Reply:
I think this solution is still better than hard coding the string, at least for English, because we then use the same message for both cases (including the same punctuation). We should talk with the team how they did i18n for DataGrid in our projects however

Reply:
We should think about localization for MUI components in general. Maybe we need to adapt our theme? https://mui.com/material-ui/guides/localization/#locale-text

Reply:
@johnnyomair Looks like setting the language in the theme dynamically isn't that easy. The correct way to do it would probably be something like MUI does here: https://mui.com/material-ui/guides/localization/#system-Locales.tsx

```ts
export default function Locales() {
  const [locale, setLocale] = React.useState<SupportedLocales>('zhCN');

  const theme = useTheme();

  const themeWithLocale = React.useMemo(
    () => createTheme(theme, locales[locale]),
    [locale, theme],
  );

  return (
      <ThemeProvider theme={themeWithLocale}>
        // ...
      </ThemeProvider>
  );
}
```

To my understanding we would have to 

- Call `useLocale()` in our `MuiThemeProvider` (which we can't atm because `useLocale()` is in cms-admin and `MuiThemeProvider` is in comet/admin)
- Match our scope to the MUI locale and dynamically import it
- Call `createCometTheme()` in the Provider
- Allow passing languages to `createCometTheme()`

So that requires some changes. I'd do it in v7

Reply:
> So that requires some changes. I'd do it in v7

Of course. I wasn't talking about this PR, but rather a general decision. Did you create a task for it?

Reply:
@johnnyomair Yes, I did